src/geo/transform.js

```diff
@@ -1535,28 +1507,33 @@ class Transform {
 
         const elevation: Elevation = this._elevation;
         this._updateCameraState();
-        const elevationAtCamera = elevation.getAtPointOrZero(this._camera.mercatorPosition);
 
-        const minHeight = this._minimumHeightOverTerrain() *  Math.cos(degToRad(this._maxPitch));
-        const terrainElevation = mercatorZfromAltitude(elevationAtCamera, this._center.lat);
+        // Find uncompensated camera position for elevation sampling.
+        // The default camera position might have been compensated by the active projection model.
```

Comment:
Just a note: this part has gotten quite complex due to conversions between different pixel coordinate units. Perhaps we could revisit this logic at some later date and do some refactoring?